internal/tcpchecker/checkport.go

```diff
@@ -0,0 +1,66 @@
+package tcpchecker
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"time"
+)
+
+func CheckPort(address string, port int, reporter Reporter) (string, error) {
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%v", address, port), 1*time.Second)
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Connect error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	err = conn.SetDeadline(time.Now().Add(1 * time.Second))
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Set deadline error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	scanner := bufio.NewScanner(conn)
+
+	output := ""
+	for scanner.Scan() {
+		output += fmt.Sprintf("%s\n", scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		if netError, ok := err.(net.Error); ok && netError.Timeout() {
+			reporter.ReportPortCheck(Report{
+				Address: address,
+				Port:    port,
+				Open:    true,
+				Status:  "Open",
```

Comment:
Should this mention a timeout instead of `Open`?

Reply:
The timeout is expected like with `nc -v -w1 ip:port`. Since the TCP is never closed, the timeout while reading is a success.

Reply:
Ah, makes sense 👍